src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -56,8 +62,13 @@ export type DropdownMenuProps<T> = {
     disabled?: boolean;
     /**
      * Menu toggle control.
+     * @deprecated Use renderSwitcher instead
      */
     switcher?: React.ReactNode;
+    /**
+     * Menu toggle control.
+     */
+    renderSwitcher?: (props: SwitcherProps) => React.ReactNode;
```

Comment:
Why we need this prop? We can keep current api, but use cloneElement to add handlers to `switcher`